### ISSUE TITLE
useModel hook & ModelProvider should warm up the model if passed modelUrl

### ISSUE THREAD

**Issue by joshuaellis** [joshuaellis/react-tensorflow#26]
## Expected Behavior

Model should be warmed up when the model is loaded with `modelUrl`. 
## Current Behavior

The model is not warmed up at all.

## Possible Solution

Could be something like the following taken from `@tensorflow/tfjs-models` –
```js
    // Warmup the model.
    const result = await this.model.executeAsync(zeroTensor) as tf.Tensor[];
    await Promise.all(result.map(t => t.data()));
    result.map(t => t.dispose());
    zeroTensor.dispose();
```

## Your Environment

- Version used:
- Browser Name and version:
- Operating System and version (desktop or mobile):
- Link to your project:

**Comment by joshuaellis** [comment]
The tensor passed to the model is indepedant to the model therefore writing a generic warmup func would be inflexible and fail on every model but one.

Therefore the approach should be accepting a function as the warmup callback & only used when `modelUrl` is used as opposed to an NPM model.

**Comment by joshuaellis** [comment]
closed with #27
